File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/Xenon.java
package com.abubusoft.xenon;

import com.abubusoft.xenon.settings.XenonSettings;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * <p>Serve a marcare i modi con i quali l'applicazione può essere lanciata.</p>
 * 
 * @author Francesco Benincasa
 * 
 */
public interface Xenon {

	/**
	 * <p>Da avviare in fase di avvio dell'applicazione</p>
	 * 
	 * @param contextValue
	 * @param settingsValue
	 */
	void init(Context contextValue, XenonSettings settingsValue);

	/**
	 * <p>Recupera l'application context</p>
	 * 
	 * @return
	 */
	Context context();
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/Xenon.kt
package com.abubusoft.xenon

import android.content.Context
import com.abubusoft.xenon.settings.XenonSettings

/**
 *
 * Serve a marcare i modi con i quali l'applicazione può essere lanciata.
 *
 * @author Francesco Benincasa
 */
interface Xenon {
    /**
     *
     * Da avviare in fase di avvio dell'applicazione
     *
     * @param contextValue
     * @param settingsValue
     */
    fun init(contextValue: Context, settingsValue: XenonSettings)

    /**
     *
     * Recupera l'application context
     *
     * @return
     */
    fun context(): Context
}